packages/core/strapi/lib/services/entity-validator/index.js

```diff
@@ -222,10 +223,116 @@ const createValidateEntity =
         entity,
       },
       { isDraft }
-    ).required();
+    )
+      .test('relations-test', 'check that all relations exist', async function (data) {
+        try {
+          await checkRelationsExist(buildRelationsStore(model.uid, data) || {});
+        } catch (e) {
+          return this.createError({
+            path: this.path,
+            message: e.message,
+          });
+        }
+        return true;
+      })
+      .required();
+
     return validateYupSchema(validator, { strict: false, abortEarly: false })(data);
   };
 
+/**
+ * Builds an object containing all the media and relations being associated with an entity
+ * @param {String} uid of the model
+ * @param {Object} data
+ * @param {Object} relationsStore to be updated and returned
+ * @returns
+ */
+const buildRelationsStore = (uid, data, relationsStore = {}) => {
+  const currentModel = strapi.getModel(uid);
+
+  Object.keys(currentModel?.attributes || {}).forEach((attributeName) => {
+    const attribute = currentModel.attributes[attributeName];
+
+    const value = data[attributeName];
+    if (isEmpty(value) || isNil(value)) {
+      return;
+    }
+
+    switch (attribute.type) {
+      case 'relation': {
+        if (!attribute.target) {
+          return;
+        }
```

Comment:
Can we remove this if? There should always be a target

Reply:
I found one case where this check is required.

in `packages/core/upload/tests/content-api/upload.test.e2e.js` - `Update a file with an entity correctly removes the relation between the entity and its old file`

collectionName `files` has the following attribute:
`related: { type: 'relation', relation: 'morphToMany', configurable: false }`

This could be the fault of the testing setup but I haven't been able to get to the bottom of it. Do you know any more?

Reply:
Update: related seems to be an attribute of `plugin::upload.file` that we will have to account for.

Reply:
Can we remove this if since the target is handled now for media type?

Reply:
I think we still need the target check as model `plugin::upload.file` has a relation `related` with no target attribute. This is separate from `attribute.type === 'media'` case

Reply:
Indeed you're right. We don't handle polymorphic relations here.
Maybe to be more explicit we can break when the relation is `morphToMany` or `morphToOne`. Wdyt? Instead of checking if the target is defined. And we can add a TODO comment to handle polymorphic relations